moveit_ros/planning/planning_pipeline/src/planning_pipeline_middleware_handle.cpp

```diff
@@ -0,0 +1,261 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2021, PickNik, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Anthony Baker */
+
+#include <moveit/planning_pipeline/planning_pipeline_middleware_handle.hpp>
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include <rclcpp/rclcpp.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <string>
+
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros_planning.planning_pipeline_middleware_handle");
+
+namespace planning_pipeline
+{
+namespace
+{
+static const auto SUBSCRIPTION_QOS = rclcpp::QoS(10);
+}
+
+PlanningPipelineMiddlewareHandle::PlanningPipelineMiddlewareHandle(const rclcpp::Node::SharedPtr& node) : node_(node)
+{
+}
+
+void PlanningPipelineMiddlewareHandle::createDisplayPathPublisher(const std::string& topic, bool flag)
+{
+  display_path_publisher_ = node_->create_publisher<moveit_msgs::msg::DisplayTrajectory>(topic, SUBSCRIPTION_QOS);
+}
+
+void PlanningPipelineMiddlewareHandle::createReceivedRequestPublisher(const std::string& topic, bool flag)
+{
+  received_request_publisher_ = node_->create_publisher<moveit_msgs::msg::MotionPlanRequest>(topic, SUBSCRIPTION_QOS);
+}
+
+void PlanningPipelineMiddlewareHandle::createContactsPublisher(const std::string& topic, bool flag)
```

Comment:
This comment applies to all the three functions above it. Is the `flag` variable actually necessary at all here? It isn't being used right now.